### PR TITLE
Font refactor

### PR DESCRIPTION
**Pull request by ZehMatt**
I think this makes a bit more sense, I also created a few new functions to burry the implementation details and how Font is added as an offset.

### CODE REVIEW

File: src/OpenLoco/src/Graphics/Gfx.cpp [hunk 1]
```diff
@@ -169,23 +181,22 @@ namespace OpenLoco::Gfx
...
+                auto* element = getG1Element(getImageIdForCharacter(font.offset, chr));
                 if (element == nullptr)
                 {
                     continue;
                 }
                 auto width = element->width + font.widthFudge;
                 // Characters from 123 to 150 are unused
                 // Unsure why this zeros it out though since a negative width isn't an issue
-                if (c >= '{' && c <= '\x96')
+                if (chr >= '{' && chr <= '\x96')
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Graphics/Gfx.cpp:195]: Oh lol i was just about to fix this before i left

File: src/OpenLoco/src/Graphics/Font.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <cstdint>
+
+namespace OpenLoco::Gfx
+{
+    enum class Font : int16_t
+    {
+        m1 = -1,
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Graphics/Font.h:9]: Could you look at all users of m1 and m2 I've a strong feeling this isn't really a font

Comment (ZehMatt) [reply src/OpenLoco/src/Graphics/Font.h:9]: It's definitely used near drawString users, however in drawString we've got code dealing with this
```cpp
if (getCurrentFont() == Font::m1)
{
    setCurrentFont(Font::medium_bold);
    _currentFontFlags = _currentFontFlags | TextDrawFlags::dark;
}
else if (getCurrentFont() == Font::m2)
{
    setCurrentFont(Font::medium_bold);
    _currentFontFlags = _currentFontFlags | TextDrawFlags::dark;
    _currentFontFlags = _currentFontFlags | TextDrawFlags::extraDark;
}
```
So m1 is bold + dark, m2 is bold + extra dark.

**Review by AaronVanGeffen — APPROVED** [review]
Looks like a solid cleanup, and all appears to be working well.

File: src/OpenLoco/src/Graphics/Font.h (hunk 2 above)
Comment (ZehMatt) [reply src/OpenLoco/src/Graphics/Font.h:9]: Yeah but not in this PR, those special attributes should be driven by the parameters.

**Pull request merged**